add: config `jwt-schema-claim-key` for schema selection in JWT (#4779)
PostgREST/postgrest: docs/references/api/schemas.rst

You can also select the schema for :ref:`functions` and :ref:`open-api`.

.. _jwt_schema_extract:

JWT Schema Extraction
~~~~~~~~~~~~~~~~~~~~~

> Schema can be specified in JWT claims. It is configured by :ref:`jwt-schema-claim-key`. This feature can be used for JWT-driven schema-based multitenancy. It allows fully hidden schema selection without exposing the schema in :ref:`profile headers <profile_headers>`. The schema specified in JWT takes precedence over profile headers.

Reviewer: A bit shorter
```suggestion
The schema can be extracted from the JWT claims using the :ref:`jwt-schema-claim-key`. This feature can be used for schema-based multitenancy. It allows hidden schema selection without exposing the schema in :ref:`profile headers <profile_headers>`. The schema specified in JWT takes precedence over profile headers.
```

Reviewer: @taimoorzaeem Sorry, I rushed into commiting this suggestion. You can squash it.

Author: No worries, that's cool.
